lib/chai/core/assertions.js

```diff
@@ -2586,24 +2589,33 @@ module.exports = function (chai, _) {
    * method or function call inside of another function. Another solution is to
    * use `bind`.
    *
-   *     expect(function () { cat.meow(); }).to.throw();  // Function expression
-   *     expect(() => cat.meow()).to.throw();             // ES6 arrow function
-   *     expect(cat.meow.bind(cat)).to.throw();           // Bind
+   *     expect(function () { cat.meow(); }).to.throw;  // Function expression
+   *     expect(() => cat.meow()).to.throw;             // ES6 arrow function
+   *     expect(cat.meow.bind(cat)).to.throw;           // Bind
    *
    * Finally, it's worth mentioning that it's a best practice in JavaScript to
-   * only throw `Error` and derivatives of `Error` such as `ReferenceError`,
-   * `TypeError`, and user-defined objects that extend `Error`. No other type of
-   * value will generate a stack trace when initialized. With that said, the
-   * `throw` assertion does technically support any type of value being thrown,
-   * not just `Error` and its derivatives.
+   * only throw `Error` instances. This includes `Error` instances created from
+   * subclassed `Error` constructors such as `ReferenceError`, `TypeError`, and
+   * user-defined constructors that extend `Error`. No other type of value will
+   * generate a stack trace when initialized. With that said, `.throw` supports
+   * any type of thrown value.
+   *
+   *     function NonErrorConstructor(message) {
+   *       this.message = message;
+   *     }
+   *     var nonErrorObject = new NonErrorConstructor('Illegal salmon!');
+   *     var badFn = function () { throw nonErrorObject; };
+   *
+   *     expect(badFn).to.throw(NonErrorConstructor, 'salmon');
+   *     expect(badFn).to.throw(new NonErrorConstructor('Illegal salmon!'));
```

Comment:
If I am to understand the proposed algorithm, the only reason this does this is because the logic falls through to using `deep-eql`. We should mention that explicitly, if it is the case, as there's plenty of edge cases around `deep-eql` which might trip a user up when making this assertion.